azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/initializer/SpringBootTelemetryInitializer.java

```diff
@@ -0,0 +1,31 @@
+package com.microsoft.applicationinsights.boot.initializer;
+
+import com.microsoft.applicationinsights.extensibility.TelemetryInitializer;
+import com.microsoft.applicationinsights.extensibility.context.DeviceContext;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ * <h1>TelemetryInitializer to set the CloudRoleName Instance</h1>
+ *
+ * <p>
+ *  This Telemetry Initializer is used to auto-configure cloud_RoleName field
+ *  to get a logical component on AppMap.
+ * </p>
+ *
+ * @author Dhaval Doshi
+ */
+public class SpringBootTelemetryInitializer implements TelemetryInitializer {
```

Comment:
I would break it into two parts:
1. Would someone like to use AI with SpringBoot without starter? : Possibility, relatively very slim though.
2. Can we move this to core? - I don't think so because we cannot distinguish if the app is of type springboot in the core and then downstream methods would fail. Further, I think this is nice separation of logics. I think spring related stuff should be at one place.